src/services/todo_service.py

```diff
@@ -0,0 +1,152 @@
+        return make_response(jsonify({"description": f"Task '{task_id}' deleted successfully"}), 200)
+
+
+def update_single_todo(task_id, change_attribute, change_content):
+    data_file = Path(Path.cwd(), "data", "data_sql.db")
+    connection = sql.connect(data_file)
+    cursor = connection.cursor()
+    if task_id_exist(data_file, task_id):
+        if change_attribute.lower() == "title":
+            statement = "UPDATE TASKS SET title = ? WHERE task_id = ?"
+            cursor.execute(statement, (change_content, task_id))
+            connection.commit()
+            cursor.close()
+            connection.close()
+            return make_response(jsonify({"description": f"Task '{task_id}' updated successfully"}), 200)
+        elif change_attribute.lower() == "description":
+            statement = "UPDATE TASKS SET description = ? WHERE task_id = ?"
+            cursor.execute(statement, (change_content, task_id))
+            connection.commit()
+            cursor.close()
+            connection.close()
+            return make_response(jsonify({"description": f"Task '{task_id}' updated successfully"}), 200)
+        elif change_attribute.lower() == "status":
+            if change_content.lower() == "to do" or change_content.lower() == "in progress" or change_content.lower() == "done":
+                statement = "UPDATE TASKS SET status = ? WHERE task_id = ?"
+                cursor.execute(statement, (change_content.lower(), task_id))
+                connection.commit()
+                cursor.close()
+                connection.close()
+                return make_response(jsonify({"description": f"Task '{task_id}' updated successfully"}), 200)
+            else:
+                return make_response(jsonify({"description": "Wrong request"}), 404)
+        elif change_attribute.lower() == "do_date":
+            statement = "UPDATE TASKS SET do_date = ? WHERE task_id = ?"
+            cursor.execute(statement, (change_content, task_id))
+            connection.commit()
+            cursor.close()
+            connection.close()
+            return make_response(jsonify({"description": f"Task '{task_id}' updated successfully"}), 200)
+        else:
+            return make_response(jsonify({"description": "Wrong request"}), 404)
+    else:
+        return make_response(jsonify({"description": "Wrong request"}), 404)
+
+
+def get_user_todos(user_id):
+    data_file = Path(Path.cwd(), "data", "data_sql.db")
+    connection = sql.connect(data_file)
+    cursor = connection.cursor()
+    statement = "SELECT * FROM TASKS WHERE user_id = ?"
+    cursor.execute(statement, (user_id,))
+    sql_data = cursor.fetchall()
+    tasks = [{"task_id": task[0], "user_id": task[1], "title": task[2], "description": task[3], "status": task[4],
+              "create_date": task[5], "do_date": task[6]} for task in sql_data]
+    if len(sql_data) == 0:
+        return make_response(jsonify({"description": "Wrong user id"}), 404)
```

Comment:
in this case I should use this:

    if len(matching_tasks) == 0:
        return make_response(jsonify({"description": "Wrong user id"}), 404)
    return make_response(jsonify(tasks), 200)

or 

    if len(matching_tasks) == 0:
        return make_response(jsonify({"description": "Wrong user id"}), 404)
    else:
        return make_response(jsonify(tasks), 200)
?

Reply:
both are ok, but I would favor first one